back/engines/commercial/report_builder/app/controllers/report_builder/web_api/v1/reports_controller.rb

```diff
@@ -7,10 +7,21 @@ class ReportsController < ::ApplicationController
         skip_before_action :authenticate_user
 
         def index
+          finder_params = {
+            text_search: params[:search],
+            owners: params[:owner_id],
+            service_reports: params[:service]
+          }
+
           reports = policy_scope(ReportBuilder::Report.global)
-          reports = paginate(reports)
+            .then { ReportBuilder::ReportFinder.new(_1, **finder_params).execute }
```

Comment:
Cool, didn't know this syntax yet

Reply:
Don't abuse it 😛 Things can get ugly real quick.